File: csharp/Lab1/bnsit/patterns/Application.cs
using System;
using Lab1.bnsit.patterns.model;

namespace Lab1.bnsit.patterns
{
	class Application
	{
		private const String PROMPT = "ees>";
		private const String HELLO_MESSAGE = "Welcome to Equipment Evidence System!";

		private ApplicationModel model = new ApplicationModel();


		private Boolean running = false;

		public void Stop()
		{
			running = false;
		}

		public void Start()
		{
			running = true;

			System.Console.WriteLine(HELLO_MESSAGE);

			while (running)
			{
				System.Console.Write(PROMPT);
				String command = System.Console.ReadLine();

				if (command.Equals("hello"))
				{
					System.Console.WriteLine(HELLO_MESSAGE);
				}
				else if (command.StartsWith("save"))
				{
					String fileName = command.Split(' ')[1];
					model.Save(fileName);
				}
				else if (command.StartsWith("load"))
				{
					String fileName = command.Split(' ')[1];
					model.Load(fileName);
				}
				else if (command.Equals("exit"))
				{
					Stop();
				}
			}
		}

		static void Main(string[] args)
		{
			Application application = new Application();
			application.Start();
		}

	}
}

File: python/start/application.py

from utils import print_, println_, readline_
from model import ApplicationModel

class Application(object):
	HELLO_MESSAGE = "Welcome to Equipment Evidence System."

	def __init__(self):
		self._running = False
		self.model = ApplicationModel() 

	def stop(self):
		self._running = False	

	def run(self):
		self._running = True

		println_(self.HELLO_MESSAGE)

		while self._running is True :
			command = input("ess > ")

			if command == "hello" :
				println_(self.HELLO_MESSAGE)

			elif command.startswith("save") :
				filename = command.split(sep = " ")[1].strip();
				self.model.save(filename)
				
			elif command.startswith("load") :
				filename = command.split(sep = " ")[1].strip();
				self.model.load(filename)

			elif command == "exit" :
				self.stop() 
				


